Resources/Prototypes/Entities/Structures/Walls/walls.yml

```diff
@@ -585,7 +555,45 @@
   - type: IconSmooth
     key: walls
     base: solid
-
+    
+- type: entity
+  parent: WallBase
+  id: WallSolidRust
+  name: solid wall
+  suffix: rusted
+  components:
+  - type: Sprite
+    sprite: Structures/Walls/solidrust.rsi
+  - type: Icon
+    sprite: Structures/Walls/solidrust.rsi
```

Comment:
Does this need an icon?

Reply:
More or less copied the entity it replaced. Is it not for wall smoothing?

Reply:
Nope, that's IconSmoothing. Icon is deprecated in most instances as appearance visualization is run on the prototype so 99% it's unnecessary now.